### ISSUE TITLE
Opensnitch custom timeout does not countdown.

### ISSUE THREAD

**Issue by ashvinlobo** [evilsocket/opensnitch#517]
Hi Gustavo...i am using Kubuntu and Ubuntu Budgie 20. From some time now i have noticed that the default timeout timer if set beyond the default time (15 seconds) does not countdown if a popup opens. This is a common issue on both the Ubuntu based OS...Ashvin

**Comment by gustavo-iniguez-goya** [comment]
Hi Ashvin,

I cannot reproduce this issue, can you post the value of `default_timeout` from `~/.config/opensnitch/settings.conf` ?

**Comment by ashvinlobo** [comment]
hi there!!!....am posting the contents of ~/.config/opensnitch/settings.conf for you as below....

[database]
file=:memory:
type=0

[global]
default_action=0
default_duration=6
default_popup_advanced=false
default_popup_advanced_dstip=false
default_popup_advanced_dstport=false
default_popup_advanced_uid=false
default_popup_position=0
default_target=0
default_timeout=20
disable_popups=false

**Comment by gustavo-iniguez-goya** [comment]
Thank you Ashvin, I guess you're using version 1.4.0. What kubuntu version did you reproduce this issue on?

I can't download the ISOs right now, I'll try it to tomorrow.

**Comment by ashvinlobo** [comment]
Yes Gustavo am using the latest version of opensnitch 1.4.0. Am using Kubuntu 20.04 and Ubuntu Budgie 21.04 on a 64 bit pc.

**Comment by ashvinlobo** [comment]
FYI ,,,the default timeout value is 15 in the latest release version and that works well but if it increase the value beyond 15 then i dont see a countdown happening on a popup.

**Comment by gustavo-iniguez-goya** [comment]
hmm, could you close the GUI, launch it from the shell and tell me if there's any warnings or errors when a pop-up appears?

**Comment by ashvinlobo** [comment]
yes...what i did was i first exited the GUI. then i checked status using systemctl which showd that Opensnitch was active. I then stopped the process using systemctl. Waited for 5 seconds and checked status then restarted the process and finally launched the GUI. Now the timeout was working and it did show me a timer countdown for a process in a popup.
Will keep you posted.

**Comment by gustavo-iniguez-goya** [comment]
okk. I just installed v1.4.0 on a fresh Ubuntu Budgie 20.04.03 installation and it worked as expected.

**Comment by ashvinlobo** [comment]
Hi Gustavo...great that it worked at your end. Do you think it's something to do with KDE Plasma 5.21?

**Comment by gustavo-iniguez-goya** [comment]
>Do you think it's something to do with KDE Plasma 5.18.5?

Nope, I've just tested it on kubuntu 20 with plasma 5.18.5 and it also works as expected.

> i have noticed that the default timeout timer if set beyond the default time (15 seconds) does not countdown if a popup opens

By the way, does the new configure timeout appear on the pop-up?

**Comment by ashvinlobo** [comment]
Yes Gustavo....for example ifi set it to 20 or 25 it does show but will be stationary.

**Comment by gustavo-iniguez-goya** [comment]
Sorry @ashvinlobo , I'm unable to reproduce this issue. I'll close it as "can't reproduce" / "works for me".

What I tried:
- Install v1.4.0 on Ubuntu Budgie 20, configure pop-ups timeout to 20 and verify that the timeout countdown works.
- Install v1.4.0 on KUbuntu 20, configure pop-ups timeout to 20 and verify that the timeout countdown works.
- Delete ~/.config/opensnitch/ and reconfigure it again.
- Logout from user's session, log in again, and see if the countdown works.

As a last try, you could place several debug messages in the file `/usr/lib/python3/dist-packages/opensnitch/dialogs/prompt.py`:
https://github.com/evilsocket/opensnitch/blob/4ea0904ff3b4356cc50a5833818d10394c80c926/ui/opensnitch/dialogs/prompt.py#L168

    def _timeout_worker(self):
        print("timeout_worker() ticks:", self._tick, "/", self._cfg.getSettings(self._cfg.DEFAULT_TIMEOUT_KEY))
        if self._tick == 0:
            self._timeout_trigger.emit()
            return

        while self._tick > 0 and self._done.is_set() is False:
            t = threading.currentThread()
            # stop only stops the coundtdown, not the thread itself.
            if getattr(t, "stop", True):
                self._tick = int(self._cfg.getSettings(self._cfg.DEFAULT_TIMEOUT_KEY))
                time.sleep(1)
                continue
            
            self._tick -= 1
            print("timeout_worker() minus one:", self._tick)
            self._tick_trigger.emit()
            time.sleep(1)

and https://github.com/evilsocket/opensnitch/blob/4ea0904ff3b4356cc50a5833818d10394c80c926/ui/opensnitch/dialogs/prompt.py#L195

    def on_tick_triggered(self):
        print("tick triggered:", self._tick)
        self._set_cmd_action_text()

and launch the GUI from the command line and see if the countdown gets stucked.

**Comment by ashvinlobo** [comment]
Hi Gustavo...sorry for being late. I tried the above steps as you suggested but when i restart the application the gui crashes. so i purged the application and reinstalled it and rebooted the workstation. All is fine but still if the timer is set above default it fails to timedown. i thinki can live with this than be a persistent headache to you :)
